byterun/minor_gc.c

```diff
@@ -548,18 +562,76 @@ void caml_empty_minor_heap_promote (struct domain* domain, int not_alone)
   caml_ev_begin("minor_gc");
   caml_ev_begin("minor_gc/remembered_set");
 
-  for (r = minor_tables->major_ref.base; r < minor_tables->major_ref.ptr; r++) {
-    oldify_one (&st, **r, *r);
+  int remembered_roots = 0;
+
+  if( not_alone ) {
+    struct domain* foreign_domain;
+    intnat cached_domains_in_minor_gc_count = atomic_load_explicit(&domains_in_minor_gc_count, memory_order_acquire);
+    // We use this rather odd scheme because it better smoothes the remainder
+    int curr_idx = domains_in_minor_gc_idx;
+    for( c = 0 ; c < cached_domains_in_minor_gc_count ; c++ ) {
+      foreign_domain = domains_in_minor_gc[c];
```

Comment:
All the domains will start at `domains_in_minor_gc[c]` and go through. Should we start at `domains_in_minor_gc_idx` to spread them out and let a domain start with it's own remembered set?